gradoop-temporal/src/main/java/org/gradoop/temporal/model/impl/TemporalGraph.java

```diff
@@ -225,6 +232,41 @@ public static TemporalGraph fromLogicalGraph(LogicalGraph logicalGraph) {
       .fromNonTemporalGraph(logicalGraph);
   }
 
+  /**
+   * Function to create a {@link TemporalGraph} from an existing {@link LogicalGraph} with valid times
+   * depending on
+   * the 3 given {@link TimeIntervalExtractor}
+   *
+   * @param logicalGraph        the existing logical Graph
+   * @param graphTimeExtractor  mapFunction to generate valid times for graphHead
+   * @param vertexTimeExtractor mapFunction to generate valid times for vertices
+   * @param edgeTimeExtractor   mapFunction to generate valid times for edges
+   * @return a temporal graph with new valid time values
+   */
+  public static TemporalGraph fromLogicalGraph(LogicalGraph logicalGraph,
+    TimeIntervalExtractor<EPGMGraphHead> graphTimeExtractor,
+    TimeIntervalExtractor<EPGMVertex> vertexTimeExtractor,
+    TimeIntervalExtractor<EPGMEdge> edgeTimeExtractor) {
+
+    TemporalGradoopConfig temporalGradoopConfig =
+      TemporalGradoopConfig.fromGradoopFlinkConfig(logicalGraph.getConfig());
+
+    DataSet<TemporalGraphHead> temporalGraphHeadDataSet = logicalGraph.getGraphHead().map(
+      new GraphHeadToTemporalGraphHead<>(
+        temporalGradoopConfig.getTemporalGraphFactory().getGraphHeadFactory(), graphTimeExtractor));
+
+    DataSet<TemporalVertex> temporalVertexDataSet = logicalGraph.getVertices().map(
+      new VertexToTemporalVertex<>(temporalGradoopConfig.getTemporalGraphFactory().getVertexFactory(),
+        vertexTimeExtractor));
+
+    DataSet<TemporalEdge> temporalEdgeDataSet = logicalGraph.getEdges().map(
+      new EdgeToTemporalEdge<>(temporalGradoopConfig.getTemporalGraphFactory().getEdgeFactory(),
+        edgeTimeExtractor));
+
+    return temporalGradoopConfig.getTemporalGraphFactory()
+      .fromDataSets(temporalGraphHeadDataSet, temporalVertexDataSet, temporalEdgeDataSet);
```

Comment:
You don't have have to do this manually. You can use the `TemporalGraphFactory` from the `TemporalGradoopConfig` directly using the `fromNonTemporalDataSets(...)` method.
```java
temporalGradoopConfig.getTemporalGraphFactory()
  .fromNonTemporalDataSets(
    logicalGraph.getGraphHead(), graphTimeExtractor,
    ...
```